module/VuFind/src/VuFind/AjaxHandler/GetItemStatuses.php

```diff
@@ -230,6 +230,22 @@ protected function reduceServices(array $rawServices)
         );
     }
 
+    /**
+     * Format call numbers and display call numbers with prefixes into a string
+     * with a split character for consumption by Javascript.
+     *
+     * @param string $prefix     Callnumber prefix or empty string.
+     * @param string $callnumber Main call number.
+     *
+     * @return string
+     */
+    protected function formatCallNums($prefix, $callnumber)
+    {
+        $displayCallnumber = !empty($prefix)
+            ? $prefix . ' ' . $callnumber : $callnumber;
+        return $callnumber . '::::' . $displayCallnumber;
```

Comment:
Might be worth adding `$separator = '::::'` as a third parameter to the function, so it's easier to override if somebody wants to use something different.

Reply:
Or is it necessary to use a separator at all? Can you just return arrays here? I think the whole response gets JSON encoded at the end, so you may be able to work with a more complex data structure. (I apologize if I'm mistaken, though -- it's been a while since I've worked with this code!)

Reply:
So, I thought about both of those options when I was working on this too... Here's why I didn't do them. The main reason was that due to the complexity of the code, I wanted to keep my footprint here as minimal as possible.

Regarding a more complex data structure, yes, everything is being JSON encoded. It's not happening here but it is happening at some point in the stack, however, everything seems to be expecting a string throughout the code, not just in the PHP but also the Javascript where this is being used. For example, here in the PHP `htmlentities` is being used on call numbers in various places and the `pickValue` method is imploding multiple call numbers into a string with a `,\t` separator as well. So, I was trying to be delicate but also keep the changes minimal. I think we can change all this but it will require more changes throughout. My fear is that there might be a reason for keeping these as strings that I'm not aware of.

Regarding the addition of the separator as a third parameter to the function, I didn't do this for two reasons, 1. that `,\t` was already being used in a prescriptive manner on call numbers and 2. (more importantly) whatever we use here must also be used in the Javascript. I thought it might make more sense to be prescriptive in this case for that reason because if someone overrides it here, they're also going to have to override it in the Javascript.

So that was my thinking when I did this. In summary, I think that adding the separator as a parameter to the function would be really easy but of less value due to the Javascript. Changing the data to an array should be doable but will require changes throughout. 

I'll defer to your judgement on how you think we should proceed.

Reply:
Thanks, @bbusenius, that all makes sense. This may be worth revisiting in the future, but I agree that for the moment, your current approach is probably the simplest and least risky.